src/java.base/share/classes/java/lang/System.java

```diff
@@ -797,6 +797,15 @@ public static native void arraycopy(Object src,  int  srcPos,
      *     <td>The module name of the initial/main module</td></tr>
      * <tr><th scope="row">{@systemProperty jdk.module.main.class}</th>
      *     <td>The main class name of the initial module</td></tr>
+     * <tr><th scope="row">{@systemProperty file.encoding}</th>
+     *     <td>The name of the default charset. Users may specify
+     *     {@code UTF-8} or {@code COMPAT} on the command line to the value.
+     *     Specifying {@code COMPAT} provides the compatible behavior with
+     *     JDKs prior to 18. In that case, the value will be replaced with the
+     *     one from {@code native.encoding} system property during runtime startup.
+     *     Specifying it to {@code UTF-8}, or no value is set, defaults to use
+     *     {@code UTF-8}. Other values have unspecified behavior.
+     *     </td></tr>
```

Comment:
Here's a suggesting re-wording to consider:

The name of the default charset, defaults to "UTF-8".  The property may be set on the command line to the value "UTF-8" or "COMPAT". If set on the command line to the value "COMPAT" then the value is replaced with the value of the native.encoding property during startup. Setting the property to a value other than "UFT-8" or "COMPAT" leads to unspecified behavior.

Reply:
Thanks, Alan. Updated the description as suggested.